### DISCUSSION TITLE
How can i detect all app exceptions, including its dependencies?

### DISCUSSION THREAD

**Discussion by wemersonrv** [adonisjs/core#2249] [Help]
Hello.

Is there a way to detect all exceptions that occur in the app, including its dependencies?

**Comment by ammezie** [comment]
If I get your question correctly, you can catch all thrown exception inside the global exception handler.

**Reply by wemersonrv** [reply]
Tx @ammezie it's exactly what i need.

**Comment by wemersonrv** [comment]
Hello again, 

The `ExceptionHandler` is for handling  HTTP request erros and in my case, i have a background service that i started it as `ioc.singleton`, and it runs a lot of stuffs in background, including queued tasks with bull... all in background and without HTTP requests... 

There is a way to adapt the `ExceptionHandler` to handle only the errors on the background and do not touch in http requests?

**Reply by thetutlage** [reply]
Since you are the one that the background service, you can simply wrap your main code (the one that starts the background service) inside a `try/catch` block and you will get all the exceptions

**Reply by wemersonrv** [reply]
Yes i use try/catch.

The problem is when working with other packages, specially when exceptions happens in realtime events insisde them and in this case i do not have a specific part on my code that i can wrap with try/catch
